compiler_opt/tools/bc_hyperparam_tuning.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for testing different hyperparameter combinations for BC training."""
+
+import os
+
+from absl import app
+from absl import flags
+import tempfile
+import gin
+
+from compiler_opt.rl import train_bc
+from compiler_opt.tools import generate_default_trace
+
+import tensorflow as tf
+from tensorboard.plugins.hparams import api as hp
+
+_DATA_PATH = flags.DEFINE_multi_string(
+    'data_path', [],
+    'Path to TFRecord file(s) containing training data. Skip training and dump'
+    'an untrained model with random weights (for testing purpose) if '
+    'unspecified.')
+_CORPUS_PATH = flags.DEFINE_string(
+    'corpus_path', '', 'Path to the corpus to evaluate the models.')
+_OUTPUT_DIR = flags.DEFINE_string('output_dir', '', 'Path to the output data')
+_GIN_FILES = flags.DEFINE_multi_string(
+    'gin_files', [], 'List of paths to gin configuration files.')
+_GIN_BINDINGS = flags.DEFINE_multi_string(
+    'gin_bindings', [],
+    'Gin bindings to override the values set in the config files.')
+
+METRIC_REWARD = 'reward'
+
+
+def run_training():
+  with tempfile.TemporaryDirectory() as temp_dir:
+    root_dir = os.path.expanduser(temp_dir)
+    root_dir = os.path.normpath(root_dir)
+    # TODO(boomanaiden154): leave num_iterations and batch_size to config
+    train_bc.train_eval(
+        num_iterations=100,
+        batch_size=64,
+        root_dir=root_dir,
+        data_path=_DATA_PATH.value)
+    return generate_default_trace.generate_default_trace(
+        data_path=_CORPUS_PATH.value,
+        policy_path=os.path.join(root_dir, './saved_policy'),
+        sampling_rate=0.01)
+
+
+def run_tuning():
+  hp_num_units_layer1 = hp.HParam('num_units_layer1', hp.Discrete([40, 80]))
+  hp_num_units_layer2 = hp.HParam('num_units_layer2', hp.Discrete([40, 80]))
```

Comment:
is it possible to organize the hp setup in a more configurable way? the code here looks very adhoc

Reply:
Yes. Definitely pretty WIP in currently. The plan now is to make everything configurable in gin so through a gin config we can set the parameters we wish to tune and the potential values within the config and then just pass that to the script where it will either just iterate through them all or potentially do something more intelligent (still need to look into libraries/methods for that a little bit more).